### DISCUSSION TITLE
Brew fails to link OpenSSL | No Github Answers working | help!

### DISCUSSION THREAD

**Discussion by rs190** [Homebrew/discussions#4129] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 3.6.20
ORIGIN: https://github.com/Homebrew/brew
HEAD: 344d32bf7f40183359592ef81ca6486e0587981b
Last commit: 4 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 983b8958109ea483ce593076e57694ca224cd308
Core tap last commit: 43 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: dodeca-core 64-bit kabylake
Clang: 14.0.0 build 1400
Git: 2.39.1 => /usr/local/bin/git
Curl: 7.84.0 => /usr/bin/curl
macOS: 13.0-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

On typing: $ brew link --force openssl

I get- 

Warning: Refusing to link macOS provided/shadowed software: openssl@3
If you need to have openssl@3 first in your PATH, run:
  echo 'export PATH="/usr/local/opt/openssl@3/bin:$PATH"' >> ~/.zshrc

For compilers to find openssl@3 you may need to set:
  export LDFLAGS="-L/usr/local/opt/openssl@3/lib"
  export CPPFLAGS="-I/usr/local/opt/openssl@3/include"

**Comment by SMillerDev** [comment]
It's not failing to link, it is refusing to link OpenSSL.

> Warning: Refusing to link macOS provided/shadowed software: openssl@3

**Reply by rs190** [reply]
Hi, what's the solution for this?

**Reply by SMillerDev** [reply]
The solution is the same as the solution to driving your car into a dolphin. It's not possible.

**Reply by rs190** [reply]
what? I require brew to link with openSSL to build a file via cmake. Any idea how I can make that work? Also, why do you say it's not possible?

**Reply by SMillerDev** [reply]
> I require brew to link with openSSL to build a file via cmake. 
> Any idea how I can make that work? 

You can tell CMake where to find OpenSSL.
> Also, why do you say it's not possible?

Because that's quite literally what the output says.
